### ISSUE TITLE
webconnectivity: do not stop processing on StatusBugNoRequests

### ISSUE THREAD

**Issue by bassosimone** [ooni/probe#1515]
Here's a minimal measurement example where it's clear stopping processing is not correct:

```JSON
{
  "input": "https://ooni.github.io",
  "software_name": "ooniprobe-android",
  "software_version": "2.7.1",
  "test_keys": {
    "agent": "redirect",
    "queries": [
      {
        "engine": "system",
        "failure": "unknown_failure: lookup ooni.github.io: No address associated with hostname",
        "hostname": "ooni.github.io",
        "query_type": "A",
        "resolver_address": "",
        "t": 0.126165039
      },
      {
        "engine": "system",
        "failure": "unknown_failure: lookup ooni.github.io: No address associated with hostname",
        "hostname": "ooni.github.io",
        "query_type": "AAAA",
        "resolver_address": "",
        "t": 0.126165039
      }
    ],
    "dns_experiment_failure": "unknown_failure: lookup ooni.github.io: No address associated with hostname",
    "dns_consistency": "inconsistent",
    "control": {
      },
      "dns": {
        "addrs": [
          "185.199.110.153",
          "185.199.111.153",
          "185.199.108.153",
          "185.199.109.153"
        ]
      }
    },
    "body_proportion": 0,
    "x_status": 16384
  },
  "test_name": "web_connectivity"
}
```

So, the bottom line here is that there are no HTTP requests but we should nonetheless flag the measurement as a failure, because this is what we get in this kind of situations. The block of code to be modified is probably:

```Go
	// Otherwise, if DNS failed with NXDOMAIN, it's DNS based blocking.
	// TODO(bassosimone): do we wanna include other errors here? Like timeout?
	if tk.DNSExperimentFailure != nil &&
		*tk.DNSExperimentFailure == errorx.FailureDNSNXDOMAINError {
		out.Accessible = &inaccessible
		out.BlockingReason = &dns
		out.Status |= StatusAnomalyDNS | StatusExperimentDNS
		return
	}
```

**Comment by hellais** [comment]
Perhaps related to: https://github.com/ooni/probe/issues/1514

**Comment by bassosimone** [comment]
Web Connectivity v0.4 produces this result for a similar test case: https://explorer.ooni.org/m/20240124140231.628922_IT_webconnectivity_514a08a20e966164. The `x_status` is not `16384` anymore, even if the `requests` entry is `nil`, rather we see `2052` (i.e., `StatusExperimentDNS | StatusSuccessNXDOMAIN`). The `accessible` and `blocking` flags are correctly set to `true` and `false` respectively. This result, though, is still wrong because the website is marked as green by explorer.

By comparison, Web Connectivity v0.5 produces: https://explorer.ooni.org/m/20240124140204.730952_IT_webconnectivity_28ec057e791f30a0. Here we have `x_null_null_flags` equal to `1` (`AnalysisFlagNullNullExpectedDNSLookupFailure`). Also, `accessible` and `blocking` are both `false`, so explorer correctly sets the nonexistent website as down.

I'm going to close this issue since it has been correctly fixed by Web Connectivity v0.5.
